### PR TITLE
[gcp] add dns resources

### PR DESCRIPTION
**Pull request by csrwng**
Adds resources for the internal DNS managed zone, its records and API record in the public zone.

Depends on https://github.com/openshift/installer/pull/1971

### CODE REVIEW

File: data/data/gcp/main.tf [hunk 1]
```diff
@@ -48,3 +48,15 @@ module "network" {
   worker_subnet_cidr = local.worker_subnet_cidr
   network_cidr       = var.machine_cidr
 }
+
+module "dns" {
+  source = "./dns"
+
+  cluster_id           = var.cluster_id
+  public_dns_zone_name = "??"
```
Comment (csrwng) [inline data/data/gcp/main.tf]: @abhinavdahiya do you have a preference for when I should get this value? 
Terraform won't let me find the public zone by domain name, so I need its name. It seems that at some point during tf var generation I'd have to reach out to GCP and get it. Should that be done here?: 
https://github.com/openshift/installer/blob/master/pkg/asset/cluster/tfvars.go#L187

Comment (csrwng) [reply data/data/gcp/main.tf]: I see we're getting it here: https://github.com/openshift/installer/blob/master/pkg/asset/manifests/dns.go#L103
Maybe there should be a common place where both (tfvars and manifests) can access it?

Comment (abhinavdahiya) [reply data/data/gcp/main.tf]: it's fine if use data source to get the public zone like we do for AWS

https://github.com/openshift/installer/blob/11aa8211d062648b20842cf5a512aabdd95bb76e/data/data/aws/route53/base.tf#L1-L3

Comment (csrwng) [reply data/data/gcp/main.tf]: The gcp data source only takes name which is not the same as the domain name:
https://www.terraform.io/docs/providers/google/d/dns_managed_zone.html

Comment (csrwng) [reply data/data/gcp/main.tf]: For now I am calling GetZone and passing it as a var:
https://github.com/openshift/installer/pull/1977/files#diff-bfad726be2a317f0c2bee99d4d62e23eR204

Comment (abhinavdahiya) [reply data/data/gcp/main.tf]: for now, it's fine.

File: data/data/gcp/dns/variables.tf [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+  type        = string
+}
+
+variable "etcd_ip_addresses" {
+  description = "List of string IPs for machines running etcd members."
+  type        = list(string)
+  default     = []
+}
+
+variable "base_domain" {
```
Comment (abhinavdahiya) [inline data/data/gcp/dns/variables.tf]: this is not used.

Comment (csrwng) [reply data/data/gcp/dns/variables.tf]: removed

**Comment by csrwng** [comment]
This should be ready to go now that #1971 has merged

File: data/data/gcp/dns/base.tf [hunk 3]
```diff
@@ -0,0 +1,51 @@
+resource "google_dns_managed_zone" "int" {
+  name       = "${var.cluster_id}-int"
+  dns_name   = "${var.cluster_domain}."
```
Comment (abhinavdahiya) [inline data/data/gcp/dns/base.tf]: do we have guarantees from the API that it will not have `.`

Comment (csrwng) [reply data/data/gcp/dns/base.tf]: The . at the end is required on create. Passing a dns name without it results in a 400 invalid dns name error.

Comment (abhinavdahiya) [reply data/data/gcp/dns/base.tf]: Bu t what if cluster_domain already has a trailing `.` ?

Comment (csrwng) [reply data/data/gcp/dns/base.tf]: ahh, ic what you mean, will fix

Comment (csrwng) [reply data/data/gcp/dns/base.tf]: @abhinavdahiya so in the terraform vars file we say we don't expect a trailing dot:
https://github.com/openshift/installer/blob/master/data/data/config.tf#L25-L32
However, when we validate install-config we do allow a trailing dot:
https://github.com/openshift/installer/blob/master/pkg/types/validation/installconfig.go#L59
Added a commit to ensure that what we pass to terraform doesn't have one.

Comment (abhinavdahiya) [reply data/data/gcp/dns/base.tf]: https://github.com/openshift/installer/pull/1977/commits/f9c69fc4c2620ebb67a8a35ec8883b52f50ceff6

doesn't solve the problem that `cluster_domain` doesn't have trailing `.` suffix

Comment (csrwng) [reply data/data/gcp/dns/base.tf]: ok *now* it's fixed

**Pull request merged**